clash_royale_client.py

```diff
@@ -12,25 +12,25 @@ def __init__(self):
         self.auth_token = os.getenv('ROYALE_API_KEY')
         self.request_headers = {'Authorization': 'Bearer {}'.format(self.auth_token)}
 
-    # url param pre-condition -> clan_tag should not have # and all caps
-    def get_clan_members(self, clan_tag):
-        cleaned_tag = clan_tag.replace('#', '').upper()
-        # Note: the '%23' is # when it is in the url
-        request_url = '{}/clans/%23{}/members'.format(self.base_url, cleaned_tag)
+    def make_request(self, request_url, method_name):
         response = requests.get(request_url, headers=self.request_headers)
         if response.status_code == 200:
             return response.json()
         else:
             # TODO is throwing an exception the best way of handling this? Should add exception handlinig at top level probs
-            raise ValueError("[get_clan_members] recieved response code: {}".format(response.status_code))
+            raise ValueError(
+                "[{}] recieved response code: {}. Error body: {}".format(method_name, response.status_code, response.json())
+            )
+
+    # url param pre-condition -> clan_tag should not have # and all caps
+    def get_clan_info(self, clan_tag):
+        cleaned_tag = clan_tag.replace('#', '').upper()
+        # Note: the '%23' is # when it is in the url
+        request_url = '{}/clans/%23{}'.format(self.base_url, cleaned_tag)

```

Comment:
I'm changing it to call the `/clans/{clanTag}` endpoint instead of `/clans/{clanTag}/members` endpoint so that we can get the clanInfo and memberList all within the same api call.